mozci/data/sources/treeherder/__init__.py

```diff
@@ -11,6 +11,7 @@
 
 from mozci.data.base import DataSource
 from mozci.errors import ContractNotFilled
+from mozci.push import Push as MozCiPush
```

Comment:
Won't this create a circular dependency?

Reply:
You could follow the same strategy as `make_push_objects` (get data from the data source, then instantiate Push objects from that data).

Reply:
Actually, we could just extend make_push_objects to be able to either use dates (as it is right now), or number of pushes

Reply:
Done :+1: